app/controllers/users_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  before_action :check_already_signed_up, only: :sign_up
+
+  VALID_USER_TYPES = %w(driver passenger).freeze
+
+  def sign_in
+    user_params = sign_in_params
+
+    user = User.find_by(email: user_params[:email])
+    raise Exceptions::BadRequest, '아이디와 비밀번호를 확인해주세요' unless user&.authenticate(user_params[:password])
+
+    log_in_user(user)
+
+    json_success({ accessToken: user.token })
+  rescue ActionController::ParameterMissing
+    render json: { message: '아이디와 비밀번호를 확인해주세요' }, status: Exceptions::BadRequest.http_status
+  end
+
+  def sign_up
+    user = CreateUserService.new(sign_up_params).run!
+
+    json_create_success(UserSerializer.new(user).as_json)
+  rescue ActionController::ParameterMissing
+    message = if !valid_email?(params[:email])
+                '올바른 이메일을 입력해주세요'
+              elsif params[:password].blank?
+                '올바른 비밀번호를 입력해주세요'
+              elsif !valid_user_type?(params[:userType])
+                '올바른 사용자 타입을 입력해주세요'
+              end
+    raise Exceptions::BadRequest, message
+  end
```

Comment:
if / else 보다는 가드문을 권장하고 사용하고 있습니다.
그리고 ParameterMissing에 대한 exception인 경우 파라미터가 누락된 경우 발생하는데 여기서는 아래 valid_xxx가 의미가 없는것 같네요.. 값에 대한 유효성 검사는 별도 메소드로 분리해주는게 좋을 것 같아요.

```suggestion
    raise Exceptions::BadRequest, '올바른 비밀번호를 입력해주세요' if params[:password].blank?
  end
    
  def sign_up_validate
    raise Exceptions::BadRequest, '올바른 이메일을 입력해주세요' unless valid_email?(params[:email])
    raise Exceptions::BadRequest, '올바른 사용자 타입을 입력해주세요' unless valid_user_type?(params[:userType])
  end
```

Reply:
[f63a93e](https://github.com/travelingaries/taxi-dispatch-backend-api/pull/1/commits/f63a93e1372d1fec0a55776eefac69a97983fb96) 작업하면서 같이 변경했습니다!